internal/aws/cwlogs/pusher.go

```diff
@@ -58,6 +59,29 @@ func NewEvent(timestampMs int64, message string) *Event {
 type StreamKey struct {
 	LogGroupName  string
 	LogStreamName string
+	Entity        *cloudwatchlogs.Entity
+}
+
+// Custom hash function for StreamKey. Necessary to uniquely identify with Entity.
+func (sk *StreamKey) Hash() string {
```

Comment:
Can you help me understand the need to include the entity in the hash? I think it would be good to include that information in the PR description as well.

Also, what are the consequences of doing this? Presumably this would result in an increase of the number of streams. Will that entail an increase in other resource usage, like goroutines?

Reply:
Can update the PR description. The main reason for having this entity included with the StreamKey (and in part, the hash), is so that we can identify which entity should be sent with the PLE call. To my understanding, even with the same LogGroupName and LogGroupStream, there could be multiple different entities that "contribute" to that stream. This was the best way that I could think of where we could still attach the correct Entity with each PLE call.

This can cause an increase in the number of StreamKeys we are tracking, and though this shouldn't increase the number of streams on the backend, it may increase the actual number of PLE calls made. I believe this will be a side-effect of this effort and was discussed earlier in conversations with Ben and Zhihong.

I think you right in that this can increase the number of goroutines.

Reply:
So we are adding a hash function because what we are adding to StreamKey is not comparable?

Reply:
In the most recent commit, we now use a bounded pusher map. It functionally behaves the same as a pusher map, but now has the size capped at 1000 pushers to prevent memory exhaustion.

> So we are adding a hash function because what we are adding to StreamKey is not comparable?

Yep, exactly.